# CRC
uint32 crc